libraries/AP_Compass/Compass_learn.cpp

```diff
@@ -246,7 +252,16 @@ void CompassLearn::process_sample(const struct sample &s)
     best_error = bestv;
     worst_error = worstv;
     best_yaw_deg = wrap_360(degrees(s.attitude.z) + besti * (360/num_sectors));
+
+    // send current learn state to gcs
+    const uint32_t now = AP_HAL::millis();
+    if (!converged && now - last_learn_progress_sent_ms >= 5000) {
+        float percent = (MIN(num_samples / COMPASS_LEARN_NUM_SAMPLES, 1.0f) + 
+                         MIN(COMPASS_LEARN_BEST_ERROR_THRESHOLD / best_error, 1.0f) + 
```

Comment:
What's the guarantee that best_error isn't 0 here?

Reply:
just use best_error+1, so we don't get a NaN, its all made up numbers anyway

Reply:
Thank you for comments. I'll fix it.